projects/kit/components/slider/slider.component.ts

```diff
@@ -104,7 +135,18 @@ export class TuiSliderComponent {
         this.el.value = `${newValue}`;
     }
 
-    protected get segmentWidth(): number {
-        return 100 / Math.max(1, this.segments);
+    protected getTicksGradient(segments: number[]): string {
+        if (segments.length <= 1) {
+            return 'linear-gradient(to right, transparent 0 100%)';
+        }
+
+        return segments.slice(1).reduce(
+            (acc, segment, index, array) =>
+                `${acc}
+                var(--tui-text-tertiary) ${segment}% calc(${segment}% + var(--t-tick-thickness)),
+                transparent ${segment}% ${array[index + 1] ?? 100}%${array[index + 1] ? ',' : ')'}
+                `,
+            `linear-gradient(to right, transparent ${segments[0]}% ${segments[1]}%,`,
```

Comment:
Should it be `0` and `segments[0]` (if we change range to `(0;1)` according to the previous comment) ?

```suggestion
            `linear-gradient(to right, transparent 0 ${segments[0]}%,`,
```